src/functions/edi/inbound/handler.ts

```diff
@@ -94,7 +94,8 @@ export const handler = async (
   } catch (e) {
     const error = ErrorWithContext.fromUnknown(e);
 
-    return failedExecution(executionId, error);
+    const failureResponse = await failedExecution(executionId, error);
+    return failureResponse;
```

Comment:
nit: since `failedExecution()` throws anyway, the return statements in each catch block can be removed (same comment applies to each of the handlers below)